ably/realtime_channel.go

```diff
@@ -47,6 +47,24 @@ func newChannels(client *Realtime) *RealtimeChannels {
 	}
 }
 
+// RTN16j, RTL15b
+func (channels *RealtimeChannels) SetChannelSerialsFromRecoverOption(serials map[string]string) {
+	for channelName, channelSerial := range serials {
+		channel := channels.Get(channelName)
+		channel.setChannelSerial(channelSerial)
+	}
+}
+
+func (channels *RealtimeChannels) GetChannelSerials() map[string]string {
+	channels.mtx.Lock()
+	defer channels.mtx.Unlock()
+	channelSerials := make(map[string]string)
+	for channelName, realtimeChannel := range channels.chans {
```

Comment:
check if channel is in `attached` state as per https://sdk.ably.com/builds/ably/specification/main/features/#RTN16g

Reply:
updated